src/jit/simdcodegenxarch.cpp

```diff
@@ -840,13 +1004,548 @@ void CodeGen::genSIMDIntrinsicUnOp(GenTreeSIMD* simdNode)
+//
+// Arguments:
+//
+// Return Value:
+//    None.
+//
+void CodeGen::genSIMDLo64BitConvert(SIMDIntrinsicID intrinsicID,
+                                    var_types       simdType,
+                                    var_types       baseType,
+                                    regNumber       tmpReg,
+                                    regNumber       tmpIntReg,
+                                    regNumber       targetReg)
+{
+    instruction ins = getOpForSIMDIntrinsic(intrinsicID, baseType);
+    if (intrinsicID == SIMDIntrinsicConvertToDouble)
+    {
+        // (Note that for mov_xmm2i, the int register is always in the reg2 position)
+        inst_RV_RV(INS_mov_xmm2i, tmpReg, tmpIntReg, TYP_LONG);
+        inst_RV_RV(ins, targetReg, tmpIntReg, baseType, emitActualTypeSize(baseType));
+    }
+    else
+    {
+        inst_RV_RV(ins, tmpIntReg, tmpReg, baseType, emitActualTypeSize(baseType));
+        inst_RV_RV(INS_mov_i2xmm, targetReg, tmpIntReg, TYP_LONG);
+    }
+}
+
+//----------------------------------------------------------------------------------
+// genSIMDIntrinsic64BitConvert: Generate code for 64-bit SIMD Convert (long/ulong <-> double)
+//
+// Arguments:
+//    simdNode - The GT_SIMD node
+//
+// Notes:
+//    There are no instructions for converting to/from 64-bit integers, so for these we
+//    do the convertion an element at a time.
+//
+void CodeGen::genSIMDIntrinsic64BitConvert(GenTreeSIMD* simdNode)
+{
+    SIMDIntrinsicID intrinsicID = simdNode->gtSIMDIntrinsicID;
+    assert((intrinsicID == SIMDIntrinsicConvertToDouble) || (intrinsicID == SIMDIntrinsicConvertToInt64) ||
+           (intrinsicID == SIMDIntrinsicConvertToUInt64));
+
+    GenTree*  op1       = simdNode->gtGetOp1();
+    var_types baseType  = simdNode->gtSIMDBaseType;
+    regNumber targetReg = simdNode->gtRegNum;
+    assert(targetReg != REG_NA);
+    var_types      simdType  = simdNode->TypeGet();
+    regNumber      op1Reg    = genConsumeReg(op1);
+    regNumber      tmpIntReg = genRegNumFromMask(simdNode->gtRsvdRegs & RBM_ALLINT);
+    regNumber      tmpReg;
+    regNumber      tmpReg2;
+    InstructionSet iset = compiler->getSIMDInstructionSet();
+
+    if (iset == InstructionSet_AVX || (baseType == TYP_ULONG))
+    {
+        // For AVX, we need two xmm tmpRegs.
+        regMaskTP tmpRegsMask = (simdNode->gtRsvdRegs & RBM_ALLFLOAT);
+        assert(tmpRegsMask != RBM_NONE);
+        assert(genCountBits(tmpRegsMask) == 2);
+
+        regMaskTP tmpReg1Mask = genFindLowestBit(tmpRegsMask);
+        tmpRegsMask &= ~tmpReg1Mask;
+        tmpReg  = genRegNumFromMask(tmpReg1Mask);
+        tmpReg2 = genRegNumFromMask(tmpRegsMask);
+    }
+    else
+    {
+        if ((intrinsicID == SIMDIntrinsicConvertToDouble) && (targetReg != op1Reg))
+        {
+            // We don't need a tmpReg; we can just use the targetReg.
+            tmpReg = targetReg;
+        }
```

Comment:
Looks like we don't need an internal register if targetReg != op1Reg.  Right now we are still reserving an internal register in all cases.

Instead we can request Register Allocator to allocate a targetReg different from its operand by marking its op1 as isDelayFree = true.  This will avoid the need for reserving an internal register.